### PR TITLE
add cov for tests

### PR DESCRIPTION
**Pull request by Charonxin**
## Why

- I used lcov to generate the current test coverage to facilitate new developers to supplement unit tests in a targeted manner.

## What is changing

- Using the xmake build tool, the corresponding flame graph can now be generated, and the command is in the comments. (Maybe it can be done using GitHub actions in the future)

## Example

image

### CODE REVIEW

**Comment by Charonxin** [comment]
image

**Comment by PikachuHy** [comment]
Thanks for your contributions.

add more detail about your PR if possible

- How our project get benefit from your PR?
- and what changed?
- give an example of how to the new feature?

**Comment by Charonxin** [comment]
1. I used lcov to generate the current test coverage to facilitate new developers to supplement unit tests in a targeted manner.

2. Using the xmake build tool, the corresponding flame graph can now be generated, and the command is in the comments. (Maybe it can be done using GitHub actions in the future)

3. cc the image above, i should exclude some path.....

**Comment by jingkaimori** [comment]
测试覆盖率只能证明测试运行的时候进入了所有分支，测试和测试覆盖率不能证明每个分支都通过了测试，因此parser的测试蒙混过关了若干次。

**Comment by PikachuHy** [comment]
LGTM.
I will merge the PR after finish Unicode support

File: xmake.lua [hunk 1]
```diff
@@ -32,6 +32,14 @@ target("pscm") do
     add_files({
         "src/**.cpp",
         "$(buildir)/version.cpp"})
+    
+    if is_mode("coverage") then
+        add_cxxflags("-O0")
+        add_cxxflags("-fprofile-arcs")
+        add_cxxflags("-ftest-coverage")
```
Comment (jingkaimori) [inline xmake.lua]: avoid multiple `add_cxxflags` invocation:
```suggestion
        add_cxxflags({
            "-O0",
            "-fprofile-arcs",
            "-ftest-coverage"})
```

File: xmake.lua [hunk 2]
```diff
@@ -62,6 +81,13 @@ for _, filepath in ipairs(os.files("test/**_tests.cpp")) do
         set_languages("cxx20")
 
         add_files(filepath)
+        
+        if is_mode ("coverage") then
+            add_cxxflags("-O0")
+            add_cxxflags("-fprofile-arcs")
+            add_cxxflags("-ftest-coverage")
```
Comment (jingkaimori) [inline xmake.lua]: same as above

**Review by jingkaimori — APPROVED** [review]
LGTM

**Comment by PikachuHy** [comment]
thanks

**Pull request merged**